lib/broccoli/ember-app.js

```diff
@@ -1081,7 +1119,54 @@ class EmberApp {
   }
 
   _addonTree(type, outputDir) {
-    let addonTrees = this.addonTreesFor(type);
+    let addonBundles = this._addonTreesFor(type);
+
+    let addonTrees = addonBundles.map(addonBundle => {
+      let addon = addonBundle.addon;
+      let tree = addonBundle.tree;
+
+      tree = this._debugTree(tree, `precompiledAddonTree:${addon.name}`);
+
+      if (!this.options.moduleNormalizerDisabled) {
+        // move legacy /modules/addon to /addon
+        let hasAlreadyPrintedModuleDeprecation;
+        tree = new ModuleNormalizer(tree, {
+          callback: () => {
+            if (!hasAlreadyPrintedModuleDeprecation) {
+              this.project.ui.writeDeprecateLine(`Addon "${addon.name}" is manually placing files in the legacy "modules" folder. Support for this will be removed in a future version.`);
```

Comment:
@rwjblue Are you sure you want addon.root in this message? I don't see what benefit that gives you over addon.name alone, since the node_modules path will be flattened, I don't think you can tell either way which addon is bringing it in if not your app.

Reply:
Without addon root you cannot detect if the message is caused by a nested addon. 

Imagine you have multiple versions of an addon. Some do not trigger the warning, but some do. This hypothetical addon is depended on heavily by many other addons. Based on the addon name alone, how can someone determine where to file an issue / PR? In this scenario the addon who’s name is printed is _already_ fixed! The actual offenders are the other addons that are including the old and deprecated version. AFAICT, this is only possible to detect via addon.root....